include/SharedMemory.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * SharedMemory.h
+ *
+ * Copyright (c) 2022 Dominic Clark <mrdomclark/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+#ifndef SHARED_MEMORY_H
+#define SHARED_MEMORY_H
+
+#include <memory>
+#include <string>
+#include <type_traits>
+
+namespace detail {
+
+class SharedMemoryImpl;
+
+class SharedMemoryData
+{
+public:
+	SharedMemoryData() noexcept;
+	SharedMemoryData(std::string&& key, bool readOnly);
+	SharedMemoryData(std::string&& key, std::size_t size, bool readOnly);
+	~SharedMemoryData();
```

Comment:
This deconstuctor does nothing given the implementation. It can be implicity defined.
```suggestion
```